dask_jobqueue/core.py

```diff
@@ -14,17 +19,108 @@ class JobQueueCluster(object):
     This class should not be used directly, use inherited class appropriate
     for your queueing system (e.g. PBScluster or SLURMCluster)
 
+    Parameters
+    ----------
+    name : str
+        Name of Dask workers.
+    threads : int
+        Number of threads per process.
+    processes : int
+        Number of processes per node.
+    memory : str
+        Bytes of memory that the worker can use. This should be a string
+        like "7GB" that can be interpretted both by PBS and Dask.
+    interface : str
+        Network interface like 'eth0' or 'ib0'.
+    death_timeout : float
+        Seconds to wait for a scheduler before closing workers
+    local_directory : str
+        Dask worker local directory for file spilling.
+    extra : str
+        Additional arguments to pass to `dask-worker`
+    kwargs : dict
+        Additional keyword arguments to pass to `LocalCluster`
+
+    Attributes
+    ----------
+    submit_command: str
+        Abstract attribute for job scheduler submit command, should be overriden
+    cancel_command: str
+        Abstract attribute for job scheduler cancel command, should be overriden
+
     See Also
     --------
     PBSCluster
     SLURMCluster
     """
-    def __init__(self):
-        raise NotImplemented
+
+    _script_template = """
+#!/bin/bash
+
+%(job_header)s
+
+%(worker_command)s
+""".lstrip()
+
+    #Following class attributes should be overriden by extending classes.
+    submit_command = None
+    cancel_command = None
+
+    def __init__(self,
+                 name='dask-worker',
+                 threads=4,
+                 processes=6,
+                 memory='16GB',
+                 interface=None,
+                 death_timeout=60,
+                 local_directory=None,
+                 extra='',
+                 **kwargs
+                 ):
```

Comment:
Can we add these options to the individual cluster doc strings?

Reply:
Not sure I get your point, where do you want to put them? They are already in the JobQueueCluster doc strings. You wan to put it in every implementation?

Reply:
My point was just that the JobQueueCluster doc string doesn't show up on the PBSCluster doc string so it requires some unnecessary searching to find all the possible arguments to the constructor.

Reply:
There was already some duplicated doc, I updated it.
Does not feel really good about that though, don't like to duplicate docstrings... Is there a better way?

Reply:
You could directly take the class docs and add to them. i.e. in each child class definition you do something like:
```
        doc = JobQueueCluster.__doc__
        self.__doc__ +=  "specific docs for Slurm cluster"
```
Perhaps there's a less hacky way though.

Reply:
@rabernat - do I recall you introducing the xarray devs to a decorator approach to doc string templating? I can't seem to dig it up but perhaps someone knows what I'm talking about.